### PR TITLE
(OraklNode) Nonce manager

### PR DESCRIPTION
**Pull request by nick-bisonai**
# Description

Basic nonce manager implementation with test code.
Will be building block for future implementation

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [x] If it is a core feature, I have added thorough tests.

## Deployment

- [ ] Should publish npm package
- [x] Should publish Docker image

### CODE REVIEW

File: node/pkg/chain/noncemanager/noncemanager.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+func (m *NonceManager) SetNonce(address string, nonce uint64) {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+
+	m.nonces[address] = nonce
+}
+
+func (m *NonceManager) GetNonceAndIncrement(address string) (uint64, error) {
```
Comment (Intizar-T) [inline node/pkg/chain/noncemanager/noncemanager.go:26]: minor suggestion to rename function to `GetAndIncrementNonce` 😄

File: node/pkg/chain/tests/noncemanager_test.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+	if nonce != 1 {
+		t.Fatalf("Expected nonce to be 1, got %d", nonce)
+	}
+
+	newNonce, err := nm.GetNonceAndIncrement(address)
+	if err != nil {
+		t.Fatalf("Expected no error, got %v", err)
+	}
+	if newNonce != 1 {
+		t.Fatalf("Expected new nonce to be 2, got %d", newNonce)
```
Comment (Intizar-T) [inline node/pkg/chain/tests/noncemanager_test.go:29]: expected to be 1?

Comment (nick-bisonai) [reply node/pkg/chain/tests/noncemanager_test.go:29]: yh the text inside t.Fatalf should be updated

**Review by Intizar-T — APPROVED** [review]
lgtm! very nice 🔥 

left some minor comments

**Pull request merged**